Remotion/Data/DomainObjects/Mapping/RelationDefinition.cs

```diff
@@ -85,10 +86,10 @@ public IRelationEndPointDefinition[] EndPointDefinitions
     {
```

Comment:
This api (+ dependencies) seems to be only used in unit tests so we should be able to remove it completely.

Reply:
Is the context at the wrong position? The `EndPointDefinitions` properties does not have anything to do with the introduction of `TypeDefinition`.

Reply:
Nope, it's this Method and related ones that also accept the `classID` as a string.
```
public IRelationEndPointDefinition? GetEndPointDefinition (string classID, string propertyName)
```
There's no production usage left, as far as I could tell.

Reply:
Are these the methods you are talking about?
![image](https://user-images.githubusercontent.com/5082673/155102197-6d4bf9c6-25ca-454e-9e11-37d260eccfe1.png)

The `GetOppositeEndPointDefinition` still has dependencies downstream and it depends on the `GetEndPointDefinition`. I don't think it makes sense to remove only parts of the API

Reply:
public IRelationEndPointDefinition? GetEndPointDefinition (string classID, string propertyName)
public IRelationEndPointDefinition? GetOppositeEndPointDefinition (string classID, string propertyName)
public bool IsEndPoint (string classID, string propertyName)

-> those 3 methods with a `string classID` parameter. The one with IRelationEndPointDefinition from the screenshot remains.

Reply:
Shouldn't we convert those into `Type, string` overloads instead? Or is there no use case for something like that? In our tests we use one of the methods but we could inline the logic there.

Reply:
Hmm... Good point. Yes, that would work and keeps the noise down.